--- ssh.py	2016-02-04 10:19:12.420625775 -0800
+++ ssh-revert_to_default_port_no_warning_message.py	2016-02-04 10:21:57.036220471 -0800
@@ -20,6 +20,7 @@
 import re
 import subprocess
 import shlex
+import socket
 import pipes
 import random
 import select
@@ -86,7 +87,13 @@
             self.common_args += ["-o", "StrictHostKeyChecking=no"]
 
         if self.port is not None:
-            self.common_args += ["-o", "Port=%d" % (self.port)]
+            test_port_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            test_port_result = test_port_socket.connect_ex( (self.host, self.port) )
+            test_port_socket.close()
+
+            if test_port_result == 0:
+                self.common_args += ["-o", "Port=%d" % (self.port)]
+
         if self.private_key_file is not None:
             self.common_args += ["-o", "IdentityFile=\"%s\"" % os.path.expanduser(self.private_key_file)]
         elif self.runner.private_key_file is not None:
