--- ssh.py	2016-02-04 07:21:27.001254830 -0800
+++ ssh.py-2016-01-28	2016-02-04 09:42:43.788449472 -0800
@@ -20,6 +20,7 @@
 import re
 import subprocess
 import shlex
+import socket
 import pipes
 import random
 import select
@@ -31,6 +32,7 @@
 from hashlib import sha1
 import ansible.constants as C
 from ansible.callbacks import vvv
+from ansible.callbacks import display
 from ansible import errors
 from ansible import utils
 
@@ -86,7 +88,15 @@
             self.common_args += ["-o", "StrictHostKeyChecking=no"]
 
         if self.port is not None:
-            self.common_args += ["-o", "Port=%d" % (self.port)]
+            test_port_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            test_port_result = test_port_socket.connect_ex( (self.host, self.port) )
+            test_port_socket.close()
+
+            if test_port_result == 0:
+                self.common_args += ["-o", "Port=%d" % (self.port)]
+            else:
+                display("SSH unable to connect to " + self.host + ":" + str(self.port) + ", reverting to default port", color='purple')
+
         if self.private_key_file is not None:
             self.common_args += ["-o", "IdentityFile=\"%s\"" % os.path.expanduser(self.private_key_file)]
         elif self.runner.private_key_file is not None:
